WordPress/src/future/res/PluginsCheckerWPOrg.java

```diff
@@ -0,0 +1,190 @@
+        HttpURLConnection conn = null;
+        String response = "";
+        try {
+            URL url = new URL(BB_PLUGINS_LIST_URL);
+            conn = (HttpURLConnection) url.openConnection();
+            conn.setReadTimeout(REQUEST_TIMEOUT_MS);
+            conn.setConnectTimeout(REQUEST_TIMEOUT_MS);
+            conn.setRequestProperty("User-Agent", USER_AGENT);
+            conn.setUseCaches(false);
+            conn.setRequestProperty("Connection", "close");
+            conn.setRequestMethod("GET");
+            BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream(), "UTF-8"));
+            String inLine;
+            while ((inLine = in.readLine()) != null) {
+                response += inLine;
+            }
+        } catch (Exception e) {
+            AppLog.e(AppLog.T.NUX, "Error while downloading the plugins list from the server...", e);
+        } finally {
+            try {
+                if (conn != null) {
+                    conn.disconnect();
+                }
+            } catch (Exception e) {
+            }
+        }
+        return response;
+    }
+
+    private int openConnection(String url) {
+        HttpURLConnection conn = null;
+        try {
+            URL requestURL = new URL(url);
+            conn = (HttpURLConnection) requestURL.openConnection();
+            conn.setReadTimeout(REQUEST_TIMEOUT_MS);
+            conn.setConnectTimeout(REQUEST_TIMEOUT_MS);
+            conn.setRequestProperty("User-Agent", USER_AGENT);
+            conn.setUseCaches(false);
+            conn.setRequestProperty("Connection", "close");
+            conn.setRequestMethod("GET");
+            conn.connect();
+            int respCode = conn.getResponseCode();
+            return respCode;
+        } catch (Exception e) {
+            AppLog.e(AppLog.T.NUX, "Error while checking for the plugins folder on the server.", e);
+        } finally {
+            try {
+                if (conn != null) {
+                    conn.disconnect();
+                }
+            } catch (Exception e) {
+            }
+
+        }
+        return 0;
+    }
+
+    private String getBaseURL(String url) {
+        String sanitizedURL = url;
+        try {
+            sanitizedURL = truncateURLAtPrefix(sanitizedURL, "wp-login.php" );
+            sanitizedURL = truncateURLAtPrefix(sanitizedURL, "/wp-admin" );
+            sanitizedURL = truncateURLAtPrefix(sanitizedURL, "/wp-content" );
+            sanitizedURL = truncateURLAtPrefix(sanitizedURL, "/xmlrpc.php" );
+        } catch (IllegalArgumentException e) {
+            AppLog.e(AppLog.T.NUX, "Can't clean the original url: " + sanitizedURL, e);
+        }
+        while (sanitizedURL.endsWith("/")) {
+            sanitizedURL = sanitizedURL.substring(0, sanitizedURL.length() - 1);
+        }
+        return sanitizedURL;
+    }
+
+    private String truncateURLAtPrefix(String url, String prefix) throws IllegalArgumentException {
```

Comment:
Yes, I noticed this, but `PluginsCheckerWPOrg` is a work in progress, not sure that the two methods will end up to be equals. Probably not. Adding just a note here, since PluginsCheckerWPOrg will be changed in the future.